### PR TITLE
Antag optin 

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Continuation of https://github.com/Skyrat-SS13/Skyrat-tg/pull/24381 under my ownership

Locks objective targetting behind "optin" settings, where the user must consent or play a particualr role to be targettable by said objective.

## How This Contributes To The Nova Sector Roleplay Experience

We're bringing back bad guys! For further details consult the policy team.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
275639567-5ab8123e-491a-4100-8d69-ad35ac93685c

</details>

### Known Bugs

- [x] Opt-In status is refreshed on decapitation and other calls of transfer_mind() - Can be fixed if I find the proc thats called when a human actually initially spawns

### Roadmap

- [x] Integration into OPFOR system to allow people to view a list of people based on their optin status

## Changelog

:cl:
add: Added antagonist opt-in! Brave crewmates wishing to be the target of nefarious evildoers now may indicate this desire in their character preferences.
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
i really mangled the merge didnt i

**Comment by nikothedude** [comment]
OH COME ON I DID IT RIGHT

**Comment by nikothedude** [comment]
why are we still ehre. just to suffeR?

**Comment by nikothedude** [comment]
I CAN MAKE IT WORK

File: code/controllers/subsystem/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -932,12 +932,16 @@
...
 	for(var/mob/living/carbon/human/player in GLOB.player_list)
 		if(!player.client || !player.mind || player.stat)
 			continue
 		if(!(player.mind.assigned_role.job_flags & JOB_CREW_MEMBER))
 			continue
-		possible_targets += player
+		// SKYRAT EDIT ADDITION START - Players in the interlink can't be obsession targets + Antag Optin
+		if (!opt_in_disabled && player.mind?.get_effective_opt_in_level() < OPT_IN_YES_ROUND_REMOVE)
+			continue
+		// SKYRAT EDIT END
```
[the remark this answers is not in the document]
Comment (nikothedude) [reply code/controllers/subsystem/dynamic/dynamic_rulesets_midround.dm:944]: how did i miss this

**Comment by Jolly-66** [comment]
You could say that typo was a minor inconvenience

**Comment by nikothedude** [comment]
https://github.com/nikothedude/GalaxiaStation/tree/optindev 

dev branch, WIP commits will be put here so things dont get pushed to live

**Comment by Zergspower** [comment]
image
is this a bug? I have antags off

**Comment by nikothedude** [comment]
kill changed to orange since yellow is invisible in lightmode

**Comment by nikothedude** [comment]
> Saw it at the end screen and I can't comment on what actually happened, but a Prisoner got targeted for an objective. i m o, given prisoners are very much a chill vibe seen nor heard role, or at least aspire to be, I think prisoners should be forced opt-in no. They exist to vibe.
> 
> Also it's not really a benefit to the _prisoner_ and just a "break into security" challenge for any would-be traitor, while disturbing the rest of the prisoners with someone else's bullshit. Big demanding lazy bottom energy fr fr

as a prisoner player, yeah, prisoners "vibe", but that doesnt mean we're unaffected by things. its not too uncommon for a window to breach or a meteor to strike or etc etc etc

i personally definitely enjoy being targeted by antags even as a prisoner, and any arguments including "but what about the other people" dont work because it can be applied to the general population of the station and thus defeat the entire point of opt in, e.g. everyone should be opt in no since it can make someone witnessing the events unhappy

also if you dont like the target, you can just ask admins for the target to be changed

**Review by GoldenAlpharex — APPROVED** [review]
Opt in deez nuts 😳

**Pull request merged**